great_expectations/datasource/new_datasource.py

```diff
@@ -45,6 +45,10 @@ def __init__(
         self._name = name
 
         self._data_context_root_directory = data_context_root_directory
+        if execution_engine is None:
+            raise ge_exceptions.ExecutionEngineError(
+                message="No ExecutionEngine configuration provided."
+            )
 
```

Comment:
If `execution_engine` being `None` causes an error to be raised, should we be setting a default value of `None` in the constructor?

Reply:
Great question, and one for @jcampbell!

Reply:
I view the answer at this point as just being that this improves the state of affairs by adding a helpful message, but isn't the long-term good state.